hera_pspec/pspecdata.py

```diff
@@ -864,44 +905,40 @@ def R(self, key):
 
         # Only add to Rkey if a particular mode is enabled
         # If you do add to this, you need to specify this in self.set_R docstring!
-        Rkey = key + (self.data_weighting,) + (self.taper,)
+        Rkey = key + (self.taper,)
         if self.data_weighting == 'dayenu':
             # add extra dayenu params
             Rkey = Rkey + tuple(self.filter_extension,) + (self.spw_Nfreqs,) \
                    + (self.symmetric_taper,)
 
+        Rkey = Rkey + (self.data_weighting,)
         if Rkey not in self._R:
             # form sqrt(taper) matrix
             if self.taper == 'none':
-                sqrtT = np.ones(self.spw_Nfreqs).reshape(1, -1)
+                myT = np.ones(self.spw_Nfreqs)
             else:
-                sqrtT = np.sqrt(dspec.gen_window(self.taper, self.spw_Nfreqs)).reshape(1, -1)
-
+                myT = dspec.gen_window(self.taper, self.spw_Nfreqs, normalization='rms')
             # get flag weight vector: straight multiplication of vectors
             # mimics matrix multiplication
-            sqrtY = np.sqrt(self.Y(key).diagonal().reshape(1, -1))
 
             # replace possible nans with zero (when something dips negative
             # in sqrt for some reason)
-            sqrtT[np.isnan(sqrtT)] = 0.0
-            sqrtY[np.isnan(sqrtY)] = 0.0
+            myT[np.isnan(myT)] = 0.0
             fext = self.filter_extension
+            nfreq = np.sum(fext) + self.spw_Nfreqs
             #if we want to use a full-band filter, set the R-matrix to filter and then truncate.
             tmat = np.zeros((self.spw_Nfreqs,
-                             self.spw_Nfreqs+np.sum(fext)),dtype=complex)
+                             nfreq), dtype=complex)
```

Comment:
lets just be direct and use np.complex128 here and below

Reply:
I've changed this to np.complex128